Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/data/Schedule.kt

```diff
@@ -0,0 +1,50 @@
+package com.mentalmachines.droidcon_boston.data
+
+
+/*
+ * View models for schedule and schedule detail items.
+ */
+class Schedule {
+
+    class ScheduleRow {
+
+        var talkDescription: String? = null
+        var talkTitle: String? = null
+        var primarySpeakerName: String? = null
+        var speakerNames: List<String>? = null
+        var speakerCount: Int = 0
+        var id: String? = null
+        var photo: String? = null
+        var utcStartTimeString: String? = null
+        var startTime: String? = null
+        var endTime: String? = null
+        var room: String? = null
+        var date: String? = null
+        var trackSortOrder: Int? = null
+
+        fun hasSpeaker(): Boolean = speakerNames != null && speakerNames!!.isNotEmpty()
+
+        fun getFirstSpeaker(): String? = if (primarySpeakerName != null) primarySpeakerName!! else speakerNames?.get(0)
```

Comment:
primarySpeakerName is always guaranteed to be non-null except for the lunch/registration "sessions"

Reply:
didn't make any assumptions - updated